backend/hitas/models/company.py

```diff
@@ -0,0 +1,137 @@
+import re
+
+from django.core.exceptions import ValidationError
+from django.db import models, transaction
+from django.utils.translation import gettext_lazy as _
+from enumfields import Enum, EnumIntegerField
+
+from hitas.models._base import ExternalHitasModel, HitasModel
+from hitas.models.codes import BuildingType, Developer, FinancingMethod, PostalCode, PublicOfficer
+from hitas.models.property_manager import PropertyManager
+
+
+def validate_business_id(value: str) -> None:
+    # Example valid value: '1234567-8'
+    match = re.search(r"^(\d{7})-(\d)$", value)
+
+    if match is None:
+        raise ValidationError(
+            _("%(value)s is not an valid business id"),
+            params={"value": value},
+        )
+
+    # TODO: verify business id with the check digit
+
+
+def validate_property_id(value: str) -> None:
+    # Example valid value: '1-1234-321-56'
+    match = re.search(r"^\d{1,4}-\d{1,4}-\d{1,4}-\d{1,4}$", value)
+
+    if match is None:
+        raise ValidationError(
+            _("%(value)s is not an valid property id"),
+            params={"value": value},
+        )
+
+
+class CompanyState(Enum):
+    NOT_READY = 0
+    LESS_THAN_30_YEARS = 1
+    GREATER_THAN_30_YEARS_NOT_FREE = 2
+    GREATER_THAN_30_YEARS_FREE = 3
+    GREATER_THAN_30_YEARS_PLOT_DEPARTMENT_NOTIFICATION = 4
+    READY_NO_STATISTICS = 5
+
+    class Labels:
+        NOT_READY = _("Not Ready")
+        LESS_THAN_30_YEARS = _("Less than 30 years")
+        GREATER_THAN_30_YEARS_NOT_FREE = _("Greater than 30 years, not free")
+        GREATER_THAN_30_YEARS_FREE = _("Greater than 30 years, free")
+        GREATER_THAN_30_YEARS_PLOT_DEPARTMENT_NOTIFICATION = _("Greater than 30 years, plot department notification")
+        READY_NO_STATISTICS = _("Ready, no statistics")
+
+
+class Company(ExternalHitasModel):
+    # Official spelling of the company name
+    official_name = models.CharField(max_length=1024)
+    # More human-friendly company name
+    display_name = models.CharField(max_length=1024)
+
+    state = EnumIntegerField(CompanyState, default=CompanyState.NOT_READY)
+    # Business ID / 'y-tunnus'
+    business_id = models.CharField(max_length=9, validators=[validate_business_id])
+
+    building_type = models.ForeignKey(BuildingType, on_delete=models.PROTECT)
+    financing_type = models.ForeignKey(FinancingMethod, on_delete=models.PROTECT)
+    public_officer = models.ForeignKey(PublicOfficer, on_delete=models.PROTECT)
+    property_manager = models.ForeignKey(PropertyManager, on_delete=models.PROTECT)
+    developer = models.ForeignKey(Developer, on_delete=models.PROTECT)
+
+    # 'hankinta-arvo'
+    acquisition_price = models.DecimalField(max_digits=15, decimal_places=2)
+    # 'toteutunut hankinta-arvo'
+    realized_acquisition_price = models.DecimalField(null=True, blank=True, max_digits=15, decimal_places=2)
+
+    # 'kiinteistötunnus'
+    property_identifier = models.CharField(max_length=19, validators=[validate_property_id])
```

Comment:
> kiinteistötunnus muodossa 091-038-0125-0018 ja tunnuksen loppuun jätetään tila tarkennukselle, tunnuksia voi olla monta (max 5)

This implementation fulfill the requirement (in HT-263). Could this maybe be moved to the `kiinteistö` model instead?

Reply:
done